src/accountingservice/main.go

```diff
@@ -93,13 +96,22 @@ func main() {
 	brokerList := strings.Split(brokers, ",")
 	log.Printf("Kafka brokers: %s", strings.Join(brokerList, ", "))
 
-	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM, syscall.SIGKILL)
 	defer cancel()
-	if err := kafka.StartConsumerGroup(ctx, brokerList, log); err != nil {
+	var consumerGroup sarama.ConsumerGroup
+	if consumerGroup, err = kafka.StartConsumerGroup(ctx, brokerList, log); err != nil {
```

Comment:
If err is not nil, then `consumerGroup` will be nil, and the `consumerGroup.Close()` in defer will panic

Reply:
In that case, the defer function is not registered because line 103 (log.Fatal(err)) exits the program